### PR TITLE
New intro

### PR DESCRIPTION
**Pull request by stephenbach**

### CODE REVIEW

File: tutorials/intro/Intro_Tutorial_1.ipynb [hunk 1]
```diff
@@ -144,41 +195,208 @@
...
+   "outputs": [],
+   "source": [
+    "from snorkel.candidates import Ngrams, CandidateExtractor\n",
+    "from snorkel.matchers import PersonMatcher\n",
+    "\n",
+    "ngrams         = Ngrams(n_max=7)\n",
+    "person_matcher = PersonMatcher(longest_match_only=True)\n",
+    "cand_extractor = CandidateExtractor(Spouse, \n",
+    "                                    [ngrams, ngrams], [person_matcher, person_matcher],\n",
+    "                                    symmetric_relations=False)"
```
Comment (ajratner) [inline tutorials/intro/Intro_Tutorial_1.ipynb]: Thoughts on setting `symmetric_relations` to `False` as default and leaving this line out?  If stays in, think it needs to be explained (has caused confusion before)

File: tutorials/intro/Intro_Tutorial_1.ipynb [hunk 2]
```diff
@@ -144,41 +195,208 @@
...
+   ],
+   "source": [
+    "from util import load_external_labels\n",
     "\n",
-    "Note that any time we execute a `UDFRunner` like `CorpusParser`, we can also execute it in parallel by running e.g.:\n",
-    "```python\n",
-    "corpus_parser.apply(doc_preprocessor, parallelism=20)\n",
-    "```\n",
-    "**Note, however, that this parallel execution will not work with SQLite, the database system used by default**; you will need to use e.g. Postgres for this!"
+    "%time load_external_labels(session, Spouse, annotator_name='gold')"
```
Comment (ajratner) [inline tutorials/intro/Intro_Tutorial_1.ipynb]: This still takes about 1.5 min for me, probably slower on many computers; does that sound correct?

**Comment by ajratner** [comment]
I can help implement the above changes once we agree on how to proceed with each of them!

**Comment by ajratner** [comment]
+ Add note at end of Intro 3 encouraging them to try training for longer (e.g. 50 epochs instead of 10) if they want to see performance go up (for me, goes from 42.5 -> 51.1 F1!)

**Comment by ajratner** [comment]
+ Also, should add a pointer to try using `GridSearch` as well (ideally, this would be in documentation and/or an "advanced" notebook, I don't know where we'd point them now, but is important / people ask about all the time)

File: README.md [hunk 3]
```diff
@@ -38,39 +38,18 @@ We're lucky to have some amazing collaborators who are currently using Snorkel!
...
-* _Data Programming with DDLite: Putting Humans in a Different Part of the Loop_, ([HILDA @ SIGMOD 2016](http://cs.stanford.edu/people/chrismre/papers/DDL_HILDA_2016.pdf))
-* _Snorkel: A System for Lightweight Extraction_, ([CIDR 2017](http://cidrdb.org/cidr2017/gongshow/abstracts/cidr2017_73.pdf))
-* Data Programming: ML with Weak Supervision ([blog](http://hazyresearch.github.io/snorkel/blog/weak_supervision.html))
-* _Learning the Structure of Generative Models without Labeled Data_, ([preprint](https://arxiv.org/abs/1703.00854))
+* _Data Programming: Creating Large Training Sets, Quickly_ ([NIPS 2016](https://papers.nips.cc/paper/6523-data-programming-creating-large-training-sets-quickly), [blog](https://hazyresearch.github.io/snorkel/blog/weak_supervision.html))
+* _Snorkel: Fast Training Set Generation for Information Extraction_ ([SIGMOD Demo 2017](http://hazyresearch.github.io/snorkel/pdfs/snorkel_demo.pdf))
+* _Data Programming with DDLite: Putting Humans in a Different Part of the Loop_ ([HILDA @ SIGMOD 2016](http://cs.stanford.edu/people/chrismre/papers/DDL_HILDA_2016.pdf))
```
Comment (stephenbach) [inline README.md]: 7f0de0a

File: tutorials/intro/Intro_Tutorial_1.ipynb [hunk 4]
```diff
@@ -144,41 +195,208 @@
...
+   "execution_count": 6,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "from snorkel.candidates import Ngrams, CandidateExtractor\n",
+    "from snorkel.matchers import PersonMatcher\n",
+    "\n",
+    "ngrams         = Ngrams(n_max=7)\n",
```
Comment (stephenbach) [inline tutorials/intro/Intro_Tutorial_1.ipynb]: Used to be 3, but was breaking up longer entities. 7 seems to cover everything.

File: tutorials/intro/Intro_Tutorial_1.ipynb (hunk 2 above)
Comment (stephenbach) [inline tutorials/intro/Intro_Tutorial_1.ipynb]: Yup, same for me.

File: tutorials/intro/Intro_Tutorial_1.ipynb [hunk 5]
```diff
@@ -144,41 +195,208 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Writing a basic `CandidateExtractor`\n",
+    "\n",
+    "Next, we'll write a basic function to extract **candidate spouse relation mentions** from the corpus.  The [Spacy](https://spacy.io/) parser we used performs _named entity recognition_ for us.\n",
+    "\n",
+    "We will extract `Candidate` objects of the `Spouse` type by identifying, for each `Sentence`, all pairs of ngrams (up to trigrams) that were tagged as people.  We do this with three objects:\n",
+    "\n",
+    "* A `ContextSpace` defines the \"space\" of all candidates we even potentially consider; in this case we use the `Ngrams` subclass, and look for all n-grams up to 3 words long\n",
```
Comment (stephenbach) [inline tutorials/intro/Intro_Tutorial_1.ipynb]: 0fd7e6f

File: tutorials/intro/Intro_Tutorial_1.ipynb (hunk 1 above)
Comment (stephenbach) [inline tutorials/intro/Intro_Tutorial_1.ipynb]: c492d46

File: tutorials/intro/Intro_Tutorial_1.ipynb [hunk 6]
```diff
@@ -144,41 +195,208 @@
...
+      "AnnotatorLabels created: 2697\n",
+      "CPU times: user 1min 41s, sys: 213 ms, total: 1min 42s\n",
+      "Wall time: 1min 42s\n"
+     ]
+    }
+   ],
+   "source": [
+    "from util import load_external_labels\n",
     "\n",
-    "Note that any time we execute a `UDFRunner` like `CorpusParser`, we can also execute it in parallel by running e.g.:\n",
```
Comment (stephenbach) [inline tutorials/intro/Intro_Tutorial_1.ipynb]: Added to advanced todo.

File: tutorials/intro/Intro_Tutorial_1.ipynb [hunk 7]
```diff
@@ -144,41 +195,208 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
```
Comment (stephenbach) [inline tutorials/intro/Intro_Tutorial_1.ipynb]: That's fine. (Doesn't bother me though, just a bit of Python :D)

I'd suggest a separate helper that takes an optional filter function as an argument.

**Pull request merged**